src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -575,5 +575,42 @@ private static IEnumerable<PropertyInfo> GetAllProperties(Type type)
 
             return allProperties;
         }
+
+        private static object GetPropertyValue(PropertyInfo property, object instance, IConfiguration config, BinderOptions options)
+        {
+            object propertyValue = property.GetValue(instance);
+            string propertyName = GetPropertyAttributeKeyName(property);
+
+            if (string.IsNullOrEmpty(propertyName))
+            {
+                propertyName = property.Name;
+            }
+
+            return BindInstance(property.PropertyType, propertyValue, config.GetSection(propertyName), options);
+        }
+
+        private static string GetPropertyAttributeKeyName(MemberInfo property)
+        {
+            if (property == null)
```

Comment:
Can this actually be null?

Reply:
Just being defensive.  Is there a reason to remove the check?

I think it's possible if the properties were in a certain protection status, but not entirely sure.

Reply:
The check is fine, but throw ArgumentNullException if you want to be defensive in this case

Reply:
Its a programming error to pass in null is my main point since this is private static

Reply:
The default case is that no property name was found. Since nothing was passed in then nothing is really being found. The downstream functions can expect the default if something wasn't found.

Including exception here makes me nervous that no exception was generated in the previous behavior.

Reply:
Maybe just refactor the helper to GetPropertyName and fully handle all cases, so it will return the property name where the caller doesn't have to deal with error cases or fixup.

```
var propertyName = GetPropertyName(property);  // handle all error cases inside
```

Reply:
Derp. Looks like handling the exception lead to failing behaviors elsewhere. Updating now.